### PR TITLE
refactor: move semantic link types into router config

### PR DESCRIPTION
**Pull request by supproduction**
Moves semantic link types into router config.

closes: [HRZ-3205](https://spryker.atlassian.net/browse/HRZ-3205)

[HRZ-3205]: https://spryker.atlassian.net/browse/HRZ-3205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/cart/entry/src/entry.component.ts [hunk 1]
```diff
@@ -105,7 +106,7 @@ export class CartEntryComponent
     if (!this.$options()?.enableItemImage) return;
 
     return html`
-      <a href=${this.$productLink()}>
+      <a href=${ifDefined(this.$productLink())}>
```
Comment (tobi-or-not-tobi) [inline libs/domain/cart/entry/src/entry.component.ts]:
```suggestion
      <a .href=${this.$productLink()}>
```

This seems to work fine, where the `ifDefined` comes for free. I'm not sure how this works, if there are some properties that will be reflected as attributes.

File: libs/domain/checkout/src/services/default-checkout.service.ts [hunk 2]
```diff
@@ -26,6 +26,7 @@ import {
   PlaceOrderStart,
   PlaceOrderSuccess,
 } from './state';
+import { RouteLinkType } from '@spryker-oryx/router/lit';
```
Comment (tobi-or-not-tobi) [inline libs/domain/checkout/src/services/default-checkout.service.ts]: can we get the type from the router entry point? Is there anything lit specific here?

File: libs/domain/search/src/models/suggestion/model.ts [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import { SemanticLinkType } from '@spryker-oryx/site';
 import { DirectiveResult } from 'lit/async-directive';
 import { SuggestionField } from '../../services';
+import { RouteLinkType } from '@spryker-oryx/router/lit';
 
 export interface SuggestionLinks {
   title?: DirectiveResult;
   options: SuggestionResource[];
-  type: SemanticLinkType;
+  type: RouteLinkType | string;
```
Comment (tobi-or-not-tobi) [inline libs/domain/search/src/models/suggestion/model.ts]: should we rename `RouteLinkType` to just `LinkType`?

File: libs/platform/experience/src/routes.ts [hunk 4]
```diff
@@ -15,27 +15,31 @@ export const defaultExperienceRoutes: RouteConfig[] = [
       route="/product/:sku"
     ></oryx-composition>`,
     name: 'Product test',
+    type: RouteLinkType.Product,
```
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/routes.ts]: Can we drop the name on these routes? Where do we use it?

Comment (tobi-or-not-tobi) [reply libs/platform/experience/src/routes.ts]: Stan I also discussed to drop the `<oryx-composition>` by default. Have you considered this?

File: libs/template/labs/src/articles/article-routes.ts [hunk 5]
```diff
@@ -1,32 +1,22 @@
...
     ></oryx-composition>`,
     name: 'FAQ',
-  },
-  {
-    path: '/article',
-    render: (): TemplateResult => html`<oryx-composition
-      route="/article"
-    ></oryx-composition>`,
-    name: 'Articles',
+    type: StoryblokContentFields.Faq,
```
Comment (tobi-or-not-tobi) [inline libs/template/labs/src/articles/article-routes.ts:8]: This route config is an example where we'd hope that the render is not needed, as we can have a fallback using the `path`. Optionally, however, one could bring in a custom render method.

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/cart/entry/src/entry.component.ts (hunk 1 above)
Comment (supproduction) [reply libs/domain/cart/entry/src/entry.component.ts]: I hope we want to keep at as attribute not as property in order of ssr. But the main point of ifDefined it's that attribute will be not rendered if value is null or undefined

File: libs/template/labs/src/articles/article-routes.ts (hunk 5 above)
Comment (supproduction) [reply libs/template/labs/src/articles/article-routes.ts:8]: But without router definition we don't know if we have this route at all

File: libs/domain/cart/entry/src/entry.component.ts (hunk 1 above)
Comment (supproduction) [reply libs/domain/cart/entry/src/entry.component.ts]: But I can drop ofc

File: libs/domain/search/src/models/suggestion/model.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply libs/domain/search/src/models/suggestion/model.ts]: mmm, is this interface even in use?

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**